### PR TITLE
Enable mininodes to construct blocks in RPC tests

### PR DESCRIPTION
**Pull request by str4d**
Includes changes to disable difficulty adjustment on regtest, cherry-picked from:
- bitcoin/bitcoin#6853
- bitcoin/bitcoin#13046

The ZIP 143/243 logic is ported from https://github.com/zcash-hackworks/zcash-test-vectors

### CODE REVIEW

**Comment by str4d** [comment]
@zkbot try

**Comment by str4d** [comment]
Note that this PR (re)introduces APIs that are not currently used, but will be in a subsequent PR.

File: qa/rpc-tests/test_framework/blocktools.py [hunk 1]
```diff
@@ -47,7 +49,7 @@ def serialize_script_num(value):
 def create_coinbase(heightAdjust = 0):
     global counter
     coinbase = CTransaction()
-    coinbase.vin.append(CTxIn(COutPoint(0, 0xffffffff), 
+    coinbase.vin.append(CTxIn(COutPoint(0, 0xffffffff),
                 CScript([counter+heightAdjust, OP_0]), 0xffffffff))
```
Comment (daira) [inline qa/rpc-tests/test_framework/blocktools.py:53]:
```suggestion
                              CScript([counter+heightAdjust, OP_0]), 0xffffffff))
```

**Review by daira — APPROVED** [review]
utACK

**Comment by daira** [comment]
@zkbot r+

**Comment by str4d** [comment]
@zkbot r-

**Comment by str4d** [comment]
@zkbot r+

**Pull request merged**